miden-lib/build.rs

```diff
@@ -58,8 +72,48 @@ fn decrease_pow(line: io::Result<String>) -> io::Result<String> {
     Ok(line)
 }
 
-/// Read and parse the contents from `./asm` into a `LibraryContents` struct, serializing it into
+fn compile_note_scripts(dst: PathBuf) -> io::Result<()> {
+    let binding = dst.join(ASM_SCRIPTS_DIR_PATH);
+    let path = Path::new(&binding);
+
+    if path.is_dir() {
+        match fs::read_dir(path) {
+            Ok(entries) => {
+                for entry in entries {
+                    match entry {
+                        Ok(file) => {
+                            let file_path = file.path();
+                            let file_path_str =
+                                file_path.to_str().unwrap_or("<invalid UTF-8 filename>");
+                            let file_name = format!(
+                                "{}.masb",
+                                file_path_str.split('/').last().unwrap().trim_end_matches(".masm")
+                            );
+                            let note_script_ast =
+                                ProgramAst::parse(&fs::read_to_string(file_path)?)?;
+                            let note_script_bytes = note_script_ast.to_bytes(AstSerdeOptions {
+                                serialize_imports: true,
+                            });
+                            fs::write(dst.join(ASL_DIR_PATH).join(file_name), note_script_bytes)?;
+                        }
+                        Err(e) => println!("Error reading directory entry: {}", e),
+                    }
+                }
+            }
+            Err(e) => println!("Error reading directory: {}", e),
+        }
+    } else {
+        println!("cargo:rerun-The specified path is not a directory.");
+    }
+
+    Ok(())
+}
+
+/// Read and parse the contents from `./asm`.
+/// - Stores contents from `/miden` into a `LibraryContents` struct, serializing it into
 /// `assets` folder under `std` namespace.
+/// - Compiles contents from `/scripts` into bytes and stores each script as `.masb`
+/// files into `assets` folder under `scripts` namespace.
```

Comment:
I think the bullet points should be:

- Compiles contents of `asm/miden` directory into a Miden library file (`.masl`) under `miden` namespace.
- Compiles contents of `asm/scripts` directory into individual `.masb` files.